internal/controller/environments/environments.go

```diff
@@ -170,6 +181,28 @@ func SetupReconcilerWithManager(
 		Complete(e)
 }
 
+func indexEnvsByApp(obj client.Object) []string {
+	env := obj.(*api.Environment) // nolint: forcetypeassert
+	if env.Spec.HealthChecks == nil {
+		return nil
+	}
+	apps := make([]string, len(env.Spec.HealthChecks.ArgoCDAppChecks))
+	for i, appCheck := range env.Spec.HealthChecks.ArgoCDAppChecks {
+		apps[i] =
+			fmt.Sprintf("%s:%s", appCheck.AppNamespace, appCheck.AppName)
+	}
+	return apps
+}
```

Comment:
See [this comment](https://github.com/akuity/kargo/pull/341/files#r1179671413)

No functional change here, but this is easier to test.